pmml-processor/src/main/java/org/springframework/cloud/stream/module/pmml/processor/PmmlProcessor.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.module.pmml.processor;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.annotation.PostConstruct;
+import javax.xml.bind.JAXBException;
+import javax.xml.transform.Source;
+
+import org.dmg.pmml.FieldName;
+import org.dmg.pmml.Model;
+import org.dmg.pmml.PMML;
+import org.jpmml.evaluator.Evaluator;
+import org.jpmml.evaluator.FieldValue;
+import org.jpmml.evaluator.ModelEvaluatorFactory;
+import org.jpmml.model.ImportFilter;
+import org.jpmml.model.JAXBUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.config.SpelExpressionConverterConfiguration;
+import org.springframework.cloud.stream.messaging.Processor;
+import org.springframework.context.annotation.Import;
+import org.springframework.expression.EvaluationContext;
+import org.springframework.expression.Expression;
+import org.springframework.expression.spel.SpelEvaluationException;
+import org.springframework.expression.spel.standard.SpelExpressionParser;
+import org.springframework.integration.annotation.ServiceActivator;
+import org.springframework.integration.context.IntegrationContextUtils;
+import org.springframework.integration.support.MutableMessage;
+import org.springframework.messaging.Message;
+import org.springframework.util.Assert;
+
+/**
+ * A processor that evaluates a machine learning model stored in PMML format.
+ *
+ * @author Eric Bottard
+ */
+@EnableBinding(Processor.class)
+@EnableConfigurationProperties(PmmlProcessorProperties.class)
+@Import({CustomConversionServiceRegistrar.class, SpelExpressionConverterConfiguration.class})
```

Comment:
Yes, we've discussed with @markfisher the benefit of having at least the `SpelExpressionConverterConfiguration` being automatically registered. I think we'll do it (but not as part of this PR obviously, several modules will need to be updated anyway)

The case for GenericMapConverter is indeed more complex. Not only is it not a `Converter<S, T>` as you mention (this bit me for a couple of hours), but it requires a reference to the `ConversionService`. And as you see in Boot code, a custom conversion service is usually created.

I propose that we merge this PR and discuss with @dsyer about how we could ease the integration of complex converter scenarios in boot.